text/0003-data-frames.md

```diff
@@ -0,0 +1,252 @@
+- Feature Name: data_frames
+- Start Date: 2020-08-05
+- RFC PR: [nushell/rfcs#0003](https://github.com/nushell/rfcs/pull/3)
+- Nushell Issue: [nushell/nushell#0000](https://github.com/nushell/nushell/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC merges the Row and Table Value types into a single new value type: Frame. Data frames take inspiration from data processing systems like R and Pandas. Data frames will play the fundamental role of modelling data in Nu and will have enough descriptive power to describe all forms of structure, including streaming tables, lists, and objects.
+
+# Motivation
+
+[motivation]: #motivation
+
+The current system has a few unexpected limitations:
```

Comment:
The inlining of nested tables is a limitation right now too, correct? If the nested table is incredibly large, we could easily run out of memory since it doesn't get streamed.

Arguably we could solve this without data frames, but it seems like what's being proposed here will potentially solve that problem?

Reply:
I'll add that to the list.

Yes, this protocol lets us stream inner tables also, so you could get the initial structure, and remember where the inner tables are, then read the contents of those inner tables from the stream.